stem/kratos_IO.py

```diff
@@ -150,6 +158,138 @@ def create_loads_process_dictionary(self, loads: List[Load]) -> Dict[str, Any]:
+        """
+
+        # initialize output dictionary
+        output_dict: Dict[str, Any] = {
+            "python_module": "json_output_process",
+            "kratos_module": "KratosMultiphysics",
+            "process_name": "JsonOutputProcess",
+            "Parameters": output.output_parameters.assemble_parameters(),
+        }
+
+        output_dict["Parameters"]["model_part_name"] = f"{DOMAIN}.{output.part_name}"
+        output_dict["Parameters"]["output_file_name"] = f"{output.output_name}"
+
+        return output_dict
+
+    def __create_output_dict(self, output: OutputProcess) -> Tuple[str, Dict[str, Any]]:
+        """
+        Creates a dictionary containing the output parameters for the desired format.
+        Allowed format are GiD, VTK and JSON.
+
+        Args:
+            output (OutputProcess): output process object
+
+        Returns:
+            str: string specifying the format of the output
+            Dict[str, Any]: dictionary containing the output parameters
+        """
+        # add output keys and parameters to dictionary based on output process type.
+        # TODO: check that keys are correct for VTK and JSON
+        if isinstance(output.output_parameters, GiDOutputParameters):
+            return "gid_output", KratosIO.__create_gid_output_dict(output=output)
+        elif isinstance(output.output_parameters, VtkOutputParameters):
+            return "vtk_output", KratosIO.__create_vtk_output_dict(output=output)
+        elif isinstance(output.output_parameters, JsonOutputParameters):
+            return "json_output", KratosIO.__create_json_output_dict(output=output)
+        else:
+            raise NotImplementedError
+
+    def create_output_process_dictionary(
+        self, outputs: List[OutputProcess]
+    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        """
+        Creates a dictionary containing the output_processes, that specifies
+        which output to request Kratos and the type of output ('GiD', 'VTK',
+        'JSON')
+
+        Args:
+            outputs (List[OutputProcess]): list of output process objects
+
+        Returns:
+            Tuple[Dict[str, Any]]: Tuple of two dictionaries containing the output
+                properties.
+                - the first containing the "output_process" dictionary. This is a
+                  separate dictionary.
+                - the second containing the "json_output" dictionary. This is to be
+                  placed under "processes".
+        """
+        output_dict: Dict[str, Any] = {"output_processes": {}}
+        json_dict: Dict[str, Any] = {"json_output": []}
+
+        for output in outputs:
+            output.output_parameters.validate()
+            key_output, _parameters_output = self.__create_output_dict(output=output)
+            if output.output_parameters.is_output_process():
+                output_dict["output_processes"][key_output] = [_parameters_output]
```

Comment:
I think this will not work in case you have multiple output processes. I.e. different outputs for different model parts. The list is always reinitialised

Reply:
Same as above, I will turn it into:

```
    "output_processes": {
        "gid_output": [
            { ... first gid output },
            { ... second gid output },
            ...
        ],
        "vtk_output": [
            { ... first vtk output },
            { ... second vtk output },
            ...
        ]
    }
```